### PR TITLE
[ADD][10.0] web_fullscreen

### PR DESCRIPTION
**Pull request by KKamaa**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Having web_responsive, do you see this interesting?

**Comment by thomaspaulb** [comment]
`web_responsive` is very cool, but pretty invasive. Customers have to really choose to go for the responsive experience. This module (ported from [v8](https://github.com/onesteinbv/addons-onestein/tree/8.0/web_fullscreen)) targets the traditional GUI and just adds a small fullscreen button on views, in order to see more clearly on small screens.

**Comment by pedrobaeza** [comment]
Ok, let's have options. Please check Travis.

File: web_fullscreen/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Onestein (<http://www.onestein.eu>)
+# Copyright 2019 Sunflower IT (<http://www.sunflowerweb.nl>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    'name': 'Fullscreen',
+    'summary': 'Adds a fullscreen mode button',
+    'author': 'Onestein ,Suflower IT, Odoo Community Association (OCA)',
```
Comment (thomaspaulb) [inline web_fullscreen/__manifest__.py]: typo, comma position

File: web_fullscreen/static/src/css/web_fullscreen.css [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.o_control_panel .o_cp_fullscreen {
+    padding-left: 5px;
+}
```
Comment (thomaspaulb) [inline web_fullscreen/static/src/css/web_fullscreen.css]: flake8

File: web_fullscreen/static/src/js/web_fullscreen.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+            /* eslint no-unused-vars: 0*/
+            this.$el.on('click', '.o_cp_fullscreen', function (e) {
+                $("#oe_main_menu_navbar").toggle();
+                $(".o_sub_menu").toggle();
+            });
+        },
+    });
+    /* eslint require-jsdoc: 0*/
+});
+
```
Comment (thomaspaulb) [inline web_fullscreen/static/src/js/web_fullscreen.js:17]: flake8

**Review by thomaspaulb — CHANGES_REQUESTED** [review]

**Review by thomaspaulb — APPROVED** [review]

**Comment by thomaspaulb** [comment]
@astirpe And here it is

File: web_fullscreen/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+# Copyright 2016 Onestein (<http://www.onestein.eu>)
+# Copyright 2019 Sunflower IT (<http://www.sunflowerweb.nl>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    'name': 'Fullscreen',
+    'summary': 'Adds a fullscreen mode button',
+    'author': 'Onestein, Sunflower IT, Odoo Community Association (OCA)',
+    'license': 'AGPL-3',
+    'website': 'http://www.onestein.eu',
```
Comment (astirpe) [inline web_fullscreen/__manifest__.py]:
```suggestion
    'website': 'https://github.com/OCA/web/',
```

File: web_fullscreen/__init__.py [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Onestein (<http://www.onestein.eu>)
```
Comment (astirpe) [inline web_fullscreen/__init__.py]: You can remove the Copyright in the `__init__.py` file.

File: web_fullscreen/README.rst [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+==========
+Fullscreen
+==========
+
+Adds a fullscreen mode button that toggles the primary
+and secondary menu.
+
```
Comment (astirpe) [inline web_fullscreen/README.rst:11]: Somewhere in the readme it should be mentioned that this module is useless if `web_responsive` is installed. Maybe you can just reuse the explanation given in https://github.com/OCA/web/pull/1226#issuecomment-476655747 :smirk: 

And nice to have: screenshots in the readme, giving a visual effect of the functionality.

**Review by astirpe — CHANGES_REQUESTED** [review]

File: web_fullscreen/static/src/xml/web_fullscreen_templates.xml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<templates xml:space="preserve">
+    <t t-extend="ControlPanel">
+        <t t-jquery=".o_cp_switch_buttons" t-operation="after">
+            <div class="btn-group btn-group-sm o_cp_fullscreen">
+                <button class="btn btn-icon fa fa-lg fa-tablet"
```
Comment (nzaterrence) [inline web_fullscreen/static/src/xml/web_fullscreen_templates.xml:6]: @KKamaa We can find a better icon for this button, still looks weird.

**Review by nzaterrence — APPROVED** [review]

**Review by astirpe — APPROVED** [review]

File: web_fullscreen/static/src/js/web_fullscreen.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+odoo.define('web.fullscreen', function (require) {
+    "use strict";
+
+    var ControlPanel = require('web.ControlPanel');
+
+    ControlPanel.include({
+        start: function () {
+            this._super.apply(this, arguments);
+            /* eslint no-unused-vars: 0*/
+            this.$el.on('click', '.o_cp_fullscreen', function (e) {
```
Comment (tarteo) [inline web_fullscreen/static/src/js/web_fullscreen.js]: Remove variable `e` and the eslint directive.

File: web_fullscreen/views/web_fullscreen_view.xml [hunk 9]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
```
Comment (tarteo) [inline web_fullscreen/views/web_fullscreen_view.xml]: change this to <odoo>

File: web_fullscreen/static/src/css/web_fullscreen.css (hunk 2 above)
Comment (tarteo) [inline web_fullscreen/static/src/css/web_fullscreen.css]: Add an extra empty line here

File: web_fullscreen/README.rst [hunk 10]
```diff
@@ -0,0 +1,47 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+==========
+Fullscreen
+==========
```
Comment (tarteo) [inline web_fullscreen/README.rst:7]: Add an extra empty line here otherwise the markup fails.

**Review by tarteo — CHANGES_REQUESTED** [review]

**Comment by tarteo** [comment]
Small remarks otherwise :+1:

**Review by tarteo — APPROVED** [review]

**Comment by tarteo** [comment]
@pedrobaeza Must the commit history be included for migrations from 8 to 10?

**Comment by pedrobaeza** [comment]
@tarteo I don't see this module in v8. Or it was in another repository?

**Comment by thomaspaulb** [comment]
Comes from [here](https://github.com/onesteinbv/addons-onestein/tree/8.0/web_fullscreen), we had permission to introduce the port here.

**Comment by pedrobaeza** [comment]
Then you must preserve commit history indeed.

**Review by pedrobaeza — DISMISSED** [review]
Preserve commit history and squash your migration commits.

**Comment by thomaspaulb** [comment]
@pedrobaeza Commits now retained, codecov error seems unrelated

**Comment by pedrobaeza** [comment]
OK, I have removed my review as it was only for that. No review on the module itself.

**Comment by thomaspaulb** [comment]
@astirpe ?

**Comment by thomaspaulb** [comment]
@pedrobaeza How many more reviews do we need?

**Comment by pedrobaeza** [comment]
OK, merging

**Comment by thomaspaulb** [comment]
Thanks!

**Pull request merged**